src/PlayerManager.cpp

```diff
@@ -563,9 +564,69 @@ void PlayerManager::exitGame(CNSocket* sock, CNPacketData* data) {
     sock->sendPacket((void*)&response, P_FE2CL_REP_PC_EXIT_SUCC, sizeof(sP_FE2CL_REP_PC_EXIT_SUCC));
 }
 
+void PlayerManager::revivePlayer(CNSocket* sock, CNPacketData* data) {
+    if (data->size != sizeof(sP_CL2FE_REQ_PC_REGEN))
+        return;
+
+    Player plr = PlayerManager::getPlayer(sock);
+
+    // players respawn at same spot they died at for now...
+    sP_CL2FE_REQ_PC_REGEN* reviveData = (sP_CL2FE_REQ_PC_REGEN*)data->buf;
+    INITSTRUCT(sP_FE2CL_REP_PC_REGEN_SUCC, response);
+    response.bMoveLocation = reviveData->eIL;
+    response.PCRegenData.iMapNum = reviveData->iIndex;
+    response.PCRegenData.iHP = 1000 * plr.level;
+    response.PCRegenData.iX = plr.x;
+    response.PCRegenData.iY = plr.y;
+    response.PCRegenData.iZ = plr.z;
+
+    sock->sendPacket((void*)&response, P_FE2CL_REP_PC_REGEN_SUCC, sizeof(sP_FE2CL_REP_PC_REGEN_SUCC));
+}
+
+void PlayerManager::enterPlayerVehicle(CNSocket* sock, CNPacketData* data) {
+    sP_CL2FE_REQ_PC_VEHICLE_ON* vehicleData = (sP_CL2FE_REQ_PC_VEHICLE_ON*)data->buf;
+    INITSTRUCT(sP_FE2CL_PC_VEHICLE_ON_SUCC, response);
+    PlayerView plrv = PlayerManager::players[sock];
+
+    // send to other players
+    //for (CNSocket* otherSock : plrv.viewable) {
+    //    otherSock->sendPacket((void*)&response, P_FE2CL_PC_VEHICLE_ON_SUCC, sizeof(sP_FE2CL_PC_VEHICLE_ON_SUCC));
```

Comment:
Hmm. There indeed doesn't seem to be a `P_FE2CL_PC_VEHICLE_ON` meant for other players. Maybe we're meant to update something in the player's aEquip and send a more generic packet? That's something to figure out later, of course.

Reply:
yeah send an equip change packet with a type and index of 8 iirc

Reply:
may want to look into iPCState in playerappearancedata, believe setting that 8 tells the client that a vehicle is equipped

Reply:
I'll take a look.

Reply:
Added iPCState into Player and sP_FE2CL_PC_EQUIP_CHANGE in the vehicle methods. These will be my final commits for today so feel free to change what ever is needed.